src/test/groovy/org/prebid/server/functional/tests/TargetingSpec.groovy

```diff
@@ -849,6 +852,211 @@ class TargetingSpec extends BaseSpec {
+        assert targeting.size() == 6
+        assert targeting.keySet().every{it -> it.startsWith(prefix)}
+    }
+
+    def "PBS auction should use long account targeting prefix when settings.targeting.truncate-attr-chars override"() {
+        given:"PBS config with setting.targeting"
+        def prefixMaxChars = PBSUtils.getRandomNumber(35, MAX_TRUNCATE_ATTR_CHARS)
+        def prebidServerService = pbsServiceFactory.getService(
+                ["settings.targeting.truncate-attr-chars":  prefixMaxChars as String])
+
+        and: "Bid request with empty targeting"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.targeting = new Targeting()
+        }
+
+        and: "Account in the DB"
+        def prefix = PBSUtils.getRandomString(prefixMaxChars - TARGETING_PREFIX_LENGTH)
+        def config = new AccountAuctionConfig(targeting: new Targeting(prefix: prefix))
+        def account = new Account(uuid: bidRequest.accountId, config: new AccountConfig(auction: config) )
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        def bidResponse = prebidServerService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response should contain default targeting prefix"
+        def targeting = bidResponse.seatbid?.first()?.bid?.first()?.ext?.prebid?.targeting
+        assert targeting.size() == 6
+        assert targeting.keySet().every{it -> it.startsWith(prefix)}
+    }
+
+    def "PBS amp should ignore and add a warning to ext.warnings when value of the account prefix is longer then settings.targeting.truncate-attr-chars"() {
+        given:"PBS config with setting.targeting"
+        def targetingChars = PBSUtils.getRandomNumber(1,10)
+        def prebidServerService = pbsServiceFactory.getService(
+                ["settings.targeting.truncate-attr-chars":  targetingChars as String])
+
+        and: "Default AmpRequest"
+        def ampRequest = AmpRequest.defaultAmpRequest
+
+        and: "Bid request"
+        def ampStoredRequest = BidRequest.defaultBidRequest
+
+        and: "Create and save stored request into DB"
+        def storedRequest = StoredRequest.getStoredRequest(ampRequest, ampStoredRequest)
+        storedRequestDao.save(storedRequest)
+
+        and: "Account in the DB"
+        def config = new AccountAuctionConfig(targeting: new Targeting(prefix: PBSUtils.getRandomString(targetingChars + 1)))
+        def account = new Account(uuid: ampRequest.account, config: new AccountConfig(auction: config) )
+        accountDao.save(account)
+
+        when: "PBS processes amp request"
+        def ampResponse = prebidServerService.sendAmpRequest(ampRequest)
+
+        then: "Amp response should contain error"
+        assert ampResponse.ext?.warnings[TARGETING]*.message == ["Key prefix value is dropped to default. " +
+                                    "Decrease custom prefix length or increase truncateattrchars by 12"]
```

Comment:
I suggest not using 12. We are based on prefix length, so to make it clear we should use random instead of `+1` and massage assert like:
```
        assert bidResponse.ext?.warnings[TARGETING]*.message == ["Key prefix value is dropped to default. " +
                                    "Decrease custom prefix length or increase truncateattrchars by ${prefix.size}"]
```